### PR TITLE
proritize given citation names as titles

### LINKED ISSUE #5209: Citation list should use names from `entities` instead of Markdown
### Is it an issue related to Adaptive Cards?

No

### Is this an accessibility issue?

No

### What version of Web Chat are you using?

Development build

### Which distribution are you using Web Chat from?

Bundle (webchat.js)

### Which hosting environment does this issue primarily affect?

Web apps

### Which browsers and platforms do the issue happened?

Others or unrelated

### Which area does this issue affect?

Chat history

### Which theme pack does this issue affect?

I did not test it on other theme packs

### What is the public URL for the website?

_No response_

### Please describe the bug

When displaying the following chat history JSON in the repro steps, it should display "Aloha" as the document name, instead of "Hello".

### Do you see any errors in console log?

_No response_

### How to reproduce the issue?

Use the following JSON to render the chat history.

```json
[
  {
    "from": {
      "role": "bot"
    },
    "type": "message",
    "timestamp": "2023-10-03T17:00:00.000Z",
    "text": "This is the document[1].\n\n[1]: _:c1 \"Hello - should not be displayed\"\n",
    "entities": [
      {
        "@context": "https://schema.org",
        "@id": "",
        "@type": "Message",
        "citation": [
          {
            "@id": "_:c1",
            "@type": "Claim",
            "appearance": {
              "@type": "DigitalDocument",
              "name": "Aloha",
              "text": "Title should be \"Aloha\" instead\n"
            },
            "position": "1"
          }
        ],
        "type": "https://schema.org/Message"
      }
    ]
  }
]
```

### What do you expect?

The document should be named "Aloha".

### What actually happened?

The document is displayed as "Hello".

### Do you have any screenshots or recordings to repro the issue?

image

### Adaptive Card JSON

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by beyackle2**
> Fixes #5209 

## Changelog Entry

-  Read deeper into citation objects in order to provide names, in PR [](), by [@beyackle2](https://github.com/beyackle2)

## Description

This lets us use the "name" field on citations as a higher-priority source for a citation name rather than falling back to what we read out of the citation text.

## Specific Changes

Adds `messageCitation.appearance.name` as the preferred default source for a name over `markdownDefinition.title`.

-  [x] I have added tests and executed them locally
- [X] I have updated `CHANGELOG.md`

## Review Checklist

> This section is for contributors to review your work.

-  [ ] Browser and platform compatibilities reviewed
-  [ ] Documents reviewed (docs, samples, live demo)
-  [ ] Internationalization reviewed (strings, unit formatting)
-  [ ] Tests reviewed (coverage, legitimacy)

### CODE REVIEW

**Review by OEvgeny — COMMENTED** [review]
LGTM :+1:

File: packages/component/src/Attachment/Text/private/MarkdownTextContent.tsx [hunk 1]
```diff
@@ -41,6 +41,10 @@ function isCitationURL(url: string): boolean {
   return onErrorResumeNext(() => new URL(url))?.protocol === 'cite:';
 }
 
+function getEntryName(entry: Entry) {
```
Comment (tdurnford) [inline packages/component/src/Attachment/Text/private/MarkdownTextContent.tsx:44]: nit: Use arrow function

Comment (beyackle2) [reply packages/component/src/Attachment/Text/private/MarkdownTextContent.tsx:44]: Good call. I'll fix this.

File: packages/component/src/Attachment/Text/private/MarkdownTextContent.tsx [hunk 2]
```diff
@@ -41,6 +41,10 @@ function isCitationURL(url: string): boolean {
...
+  return entry.claim?.appearance?.usageInfo?.name;
```
Comment (tdurnford) [inline packages/component/src/Attachment/Text/private/MarkdownTextContent.tsx:45]: Why only add a function for `getEntryName` and not things like `getEntryTitle` or `getEntryDescription`?

Comment (tdurnford) [inline packages/component/src/Attachment/Text/private/MarkdownTextContent.tsx:45]: Should this account for other possible names?

```suggestion
  return entry.claim?.name ?? entry.claim?.appearance?.usageInfo?.name;
```

Comment (beyackle2) [reply packages/component/src/Attachment/Text/private/MarkdownTextContent.tsx:45]: Despite the name of that field and this function, I think `entry.claim?.name` and `entry.claim?.appearance?.usageInfo?.name` are different things. I'll see if I can at put some explanatory comments on this.

File: packages/component/src/LinkDefinition/private/ItemBody.tsx [hunk 3]
```diff
@@ -4,11 +4,20 @@ import Badge from './Badge';
...
-  badgeName?: string;
-  badgeTitle?: string;
+  // The text (usually a number) displayed at the head of the citation
   identifier?: string;
-  isExternal?: boolean;
+
+  // The text displayed as the main link of the citation
   text: string;
+
+  // do we show the external link icon after the link?
```
Comment (tdurnford) [inline packages/component/src/LinkDefinition/private/ItemBody.tsx:13]: Should probably answer this before merging.

Comment (beyackle2) [reply packages/component/src/LinkDefinition/private/ItemBody.tsx:13]: That's a description of the boolean field. I'll make that clearer.

**Review by tdurnford — APPROVED** [review]

**Pull request merged**